src/tools/imgtool/library.cpp

```diff
@@ -128,155 +110,124 @@ void imgtool_library_add(imgtool_library *library, imgtool_get_info get_info)
 		i = 0;
 		do
 		{
-			/* clear out the class */
+			// clear out the class
 			memset(&imgclass, 0, sizeof(imgclass));
 			imgclass.get_info = get_info;
 
-			/* make the class */
+			// make the class
 			result = make_class(i++, &imgclass);
 			if (result)
-				imgtool_library_add_class(library, &imgclass);
+				add_class(&imgclass);
 		}
 		while(result);
 	}
 	else
 	{
-		imgtool_library_add_class(library, &imgclass);
+		add_class(&imgclass);
 	}
 }
 
 
+//-------------------------------------------------
+//  unlink
+//-------------------------------------------------
 
-const imgtool_module *imgtool_library_unlink(imgtool_library *library,
-	const char *module)
+void library::unlink(const char *module_name)
 {
-	imgtool_module *m;
-	imgtool_module **previous;
-	imgtool_module **next;
-
-	for (m = library->first; m; m = m->next)
-	{
-		if (!core_stricmp(m->name, module))
-		{
-			previous = m->previous ? &m->previous->next : &library->first;
-			next = m->next ? &m->next->previous : &library->last;
-			*previous = m->next;
-			*next = m->previous;
-			m->previous = nullptr;
-			m->next = nullptr;
-			return m;
-		}
-	}
-	return nullptr;
+	modulelist::iterator iter = find(module_name);
+	if (iter != m_modules.end())
+		m_modules.erase(iter);
 }
 
 
+//-------------------------------------------------
+//  module_compare
+//-------------------------------------------------
 
-static int module_compare(const imgtool_module *m1,
-	const imgtool_module *m2, imgtool_libsort_t sort)
+int library::module_compare(const imgtool_module *m1, const imgtool_module *m2, sort_type sort)
 {
 	int rc = 0;
 	switch(sort)
 	{
-		case ITLS_NAME:
-			rc = strcmp(m1->name, m2->name);
-			break;
-		case ITLS_DESCRIPTION:
-			rc = core_stricmp(m1->name, m2->name);
-			break;
+	case sort_type::NAME:
+		rc = strcmp(m1->name, m2->name);
+		break;
+	case sort_type::DESCRIPTION:
+		rc = core_stricmp(m1->description, m2->description);
+		break;
 	}
 	return rc;
 }
 
 
+//-------------------------------------------------
+//  sort
+//-------------------------------------------------
 
-void imgtool_library_sort(imgtool_library *library, imgtool_libsort_t sort)
+void library::sort(sort_type sort)
 {
-	imgtool_module *m1;
-	imgtool_module *m2;
-	imgtool_module *target;
-	imgtool_module **before;
-	imgtool_module **after;
-
-	for (m1 = library->first; m1; m1 = m1->next)
+	auto compare = [&](const std::unique_ptr<imgtool_module> &a, const std::unique_ptr<imgtool_module> &b)
```

Comment:
Please don't use auto-catpture.  Explicitly capture `this` and `sort` to make it clear what the lambda is copying.